pkg/util/permissions/permissions_test.go

```diff
@@ -0,0 +1,74 @@
+package permissions
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/authorization/mgmt/2015-07-01/authorization"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func TestCanDoAction(t *testing.T) {
+	tests := []struct {
+		name        string
+		permissions []authorization.Permission
+		action      string
+		want        bool
+		wantErr     bool
+	}{
+		{
+			name:        "empty permissions list",
+			permissions: []authorization.Permission{},
+			action:      "Microsoft.Network/virtualNetworks/subnets/join/action",
+			want:        false,
+		},
+		{
+			name: "has permission",
+			permissions: []authorization.Permission{
+				{
+					Actions:    to.StringSlicePtr([]string{"Microsoft.Compute/virtualMachines/*"}),
+					NotActions: to.StringSlicePtr([]string{}),
+				},
+				{
+					Actions:    to.StringSlicePtr([]string{"Microsoft.Network/virtualNetworks/subnets/join/action"}),
+					NotActions: to.StringSlicePtr([]string{}),
+				},
+			},
+			action: "Microsoft.Network/virtualNetworks/subnets/join/action",
+			want:   true,
+		},
+		{
+			name: "has permission excluded - exact",
+			permissions: []authorization.Permission{{
+				Actions:    to.StringSlicePtr([]string{"Microsoft.Network/*"}),
+				NotActions: to.StringSlicePtr([]string{"Microsoft.Network/virtualNetworks/subnets/join/action"}),
+			}},
+			action: "Microsoft.Network/virtualNetworks/subnets/join/action",
+			want:   false,
+		},
+		{
+			name: "has permission excluded - wildcard",
+			permissions: []authorization.Permission{{
```

Comment:
one more test case would be good: that of 2 permissions, one which allows, the second which denies via a NotAction -- the result should be allow